### PR TITLE
Fix bigquery copy materialization

### LINKED ISSUE #820: [ADAP-695] [Regression] Error when use REF inside COPY materialization
### Is this a regression in a recent version of dbt-bigquery?

- [X] I believe this is a regression in dbt-bigquery functionality
- [X] I have searched the existing issues, and I could not find an existing issue for this regression

### Current Behavior

if you use REF in model with copy materialization, then you receive following error:

```
Compilation Error in model bla-bla
Model 'bla-bla' (models/model_copy.sql) depends on a node named 'name' which was not found

> in macro materialization_copy_bigquery (macros/materializations/copy.sql)
> called by model model_copy (models/model_copy.sql)

```

I've checked all previous versions and there is no issue at all.
the issue is on 1.5 only
if you use SOURCE then all works fine
only REF usage leads above issue

### Expected/Previous Behavior

both REF and SOURCE works fine within COPY materialization

### Steps To Reproduce

use copy materialization with any of your source or model.
config exampe:

```
{{ config(
    materialized='copy'
)}}

select * from {{ source('ANY_YOUR_SOURCE') }}, {{ ref('ANY_YOUR_MODEL') }}

```

if you run this model on 1.5.x then you receive an error
then delete REF (or change it to any other SOURCE) from this model and run. All works fine

### Relevant log output

_No response_

### Environment

```markdown
- OS: dockerized env using `python:3.11-slim` image
- Python: 3.11
- dbt-core (working version): 1.4.6
- dbt-bigquery (working version): 1.4.3
- dbt-core (regression version): 1.5.2 (all 1.5.x)
- dbt-bigquery (regression version): 1.5.3 (all 1.5.x)
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by m-sche**
resolves #820

### Problem

Since dbt 1.5, bigquery's `materialization='copy'` does not work due to changed structure of `model.refs` and the introduction of model versioning. It used to be an array of arrays in 1.4, since 1.5 it is an array of dictionaries.

The `ref` macro currently accepts one mandatory positional argument (name), one optional positional argument (package), and one optional keyword argument (version).

Model versioning docs: https://docs.getdbt.com/docs/collaborate/govern/model-versions
Ref macro arguments docs: https://docs.getdbt.com/reference/dbt-jinja-functions/ref#advanced-ref-usage

### Solution

This PR adapts to the new structure of `model.refs` and allows for an optional version argument.

A functional test has been added to test if a very simple copy/incremental materialization works. This test should catch similar issues in the future. The test fails if executed against the current code and passes when executed against the fixed code in this PR. The test is heavily inspired by `test_simple_copy.sql` test from this repository.

A similar solution was discussed in #820 but the discussion got stale, hence this PR, but I am keen on receiving any feedback!

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by m-sche** [comment]
Added a changie entry and fixed an account from which the commits were created

**Comment by m-sche** [comment]
From my point of view, this PR is ready for review, please let me know if there's anything missing :)

File: .changes/unreleased/Fixes-20230906-141819.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Fixes
+body: 'Fix bigquery copy materialization #910'
+time: 2023-09-06T14:18:19.445262+02:00
+custom:
+  Author: m-sche
+  Issue: '#820'
```
Comment (dbeatty10) [inline .changes/unreleased/Fixes-20230906-141819.yaml:6]: Changing the formatting for the changelog so the hyperlinks work, etc.
```suggestion
kind: Fixes
body: Fix bigquery copy materialization
time: 2023-09-06T14:18:19.445262+02:00
custom:
  Author: m-sche
  Issue: "820"
```

**Review by mikealfare — APPROVED** [review]

**Comment by dbeatty10** [comment]
@mikealfare and @dataders:

I just added backport labels for 1.5 and 1.6 per https://github.com/dbt-labs/dbt-bigquery/issues/820#issuecomment-1753121110.

What else do we need to do to create backport PRs for these?

**Pull request merged**